core/src/main/java/org/infinispan/partitionhandling/impl/PartitionHandlingManager.java

```diff
@@ -20,5 +28,87 @@
 
    void checkBulkRead();
 
+   @Deprecated
+      //test use only. it can be removed if we update the tests
    CacheTopology getLastStableTopology();
+
+   /**
+    * Adds a partially aborted transaction.
+    * <p/>
+    * The transaction should be registered when it is not sure if the abort happens successfully in all the affected
+    * nodes. It can rethrow the exception is the transaction is not registered.
+    *
+    * @param globalTransaction the global transaction.
+    * @param affectedNodes     the nodes involved in the transaction and they must abort the transaction.
+    * @param lockedKeys        the keys locally locked.
+    * @param suspectException  the suspected exception. It contains the node who left the view.
+    */
+   void addPartialRollbackTransaction(GlobalTransaction globalTransaction, Collection<Address> affectedNodes,
+                                      Collection<Object> lockedKeys, SuspectException suspectException);
```

Comment:
We've already talked about this on IRC, but I'll answer here as well. The idea is that we don't know whether we're going to stay available or not based on just the SuspectException, we need to wait for a new topology from the coordinator. If our partition stayed available even without the leavers, that means we can retry on the new owners and the commit will succeed. If our partition became degraded, the ISPN-5274 solution is to throw an AvailabilityException even if the transaction was committed on some (or all) of the owners.